### ISSUE TITLE
False positive unnecessary ellipses for parameters with ellipses as a default value when a docstring is present

### ISSUE THREAD

**Issue by BradLucky** [PyCQA/pylint#7134]
### Bug description

When setting a default argument of `...`, pylint claims that as `unnecessary-ellipsis`, even though that would change the behavior of the method and thus is not unnecessary. It only affects methods with docstrings.
```python
def my_func(foo=...):
    return foo  # emits nothing

def my_doc_func(foo=...):
    """docstring"""
    return foo  # pylint emits: `test.py:4:20: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)`
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:20: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)

-----------------------------------
Your code has been rated at 7.50/10
```

### Expected behavior

No warning for ellipsis used as a default argument.

### Pylint version

```shell
pylint 2.13.0
astroid 2.11.3
Python 3.9.6 (default, Sep 21 2021, 00:31:42)
[GCC 7.3.1 20180712 (Red Hat 7.3.1-13)]
```

**Comment by mbyrnepr2** [comment]
Please upgrade to the latest Pylint where this issue no longer occurs; no longer over-zealous & no PR necessary :D

**Comment by Pierre-Sassoulas** [comment]
Thanks @mbyrnepr2, I should have tried reproducing :)
